### PR TITLE
Added AccountMapper Test class

### PR DESCRIPTION
**Pull request by Arturnix**
Test to check is correct account fetched into data model

### CODE REVIEW

File: src/test/java/AccountMapperTest.java [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
-   // @Mock
+   @Mock
     private FetchToken testFetchToken = new FetchToken();
-   // @Mock
+   @Mock
     private AccountMapper testAccountMapper = new AccountMapper();
 
    @Test
-    public void correctFetchedAccountToDataModelTest() {
+    public void isCorrectAccountFetchedToDataModelTest() { // czy obsługiwac w testach zgodnosc danych pobranych jako pozostale pola w data modelu?
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:28]: please add negative paths

Comment (Arturnix) [reply src/test/java/AccountMapperTest.java]: Rest of happy paths added.
Negative paths added.

File: src/main/java/com/diablo3CharViewer/token/FetchToken.java [hunk 2]
```diff
@@ -91,9 +91,9 @@ public String fetchAPIResourceRequest(String battleNETURL) {
             HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
 
             if(response.statusCode() != HttpURLConnection.HTTP_OK) {
-                return "Bohater o podanym battleTagu nie istanieje w swiecie Sanktuarium"; //System.out.println("Bohater o podanym battleTagu nie istanieje w swiecie Sanktuarium");
-                //throw new RuntimeException("HttpResponseCode: " + response.statusCode());
-                //throw new RuntimeException("Bohater o podanym battleTagu nie istanieje w swiecie Sanktuarium");
+               //return "Bohater o podanym battleTagu nie istnieje w swiecie Sanktuarium"; //System.out.println("Bohater o podanym battleTagu nie istnieje w swiecie Sanktuarium!");
+                throw new RuntimeException("Bohater o podanym battleTagu nie istnieje w swiecie Sanktuarium!\n" + "HttpResponseCode: " + response.statusCode());
```
Comment (blazejprzyluski) [inline src/main/java/com/diablo3CharViewer/token/FetchToken.java:95]: Bohater czy konto?

File: src/test/java/AccountMapperTest.java [hunk 3]
```diff
@@ -3,30 +3,131 @@
...
+
+        mapKills.put("elites", 1974);
+
+        lenient().when(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken)).thenReturn(accountDataModel);
+        Assertions.assertTrue(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getKills().containsKey("elites"));
+        Assertions.assertTrue(testAccountMapper.fetchAccountToDataModel(battleTag, testFetchToken).getKills().containsValue(1974));
+    }
+
+    @Test
+    public void wrongBattleTagFormatProvided() {
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:88]: In this method what you're doing is setting a String to be returned and then checking if the string is the same you put in, which is not really checking

File: src/test/java/AccountMapperTest.java [hunk 4]
```diff
@@ -3,30 +3,131 @@
...
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void providedBattleTagIsNullThrowsException() {
 
-        String battleTag = "Jokefish#2265";
+        AccountMapper testAccountMapper = new AccountMapper();
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:120]: Why the new instances here?

File: src/test/java/AccountMapperTest.java [hunk 5]
```diff
@@ -3,30 +3,131 @@
...
+        Mockito.when(testAccountMapper.generateRequest(providedBattleTag, testFetchToken))
+                .thenReturn("Niepoprawny format battleTag! Spróbuj ponownie.");
+
+        Assertions.assertEquals(wrongBattleTagFormatWarning, testAccountMapper.generateRequest(providedBattleTag, testFetchToken));
+    }
+
+    @Test
+    public void providedBattleTagDoesntExistThrowsException() {
+
+        AccountMapper testAccountMapper = new AccountMapper();
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:102]: Why the new instances here?

File: src/test/java/AccountMapperTest.java [hunk 6]
```diff
@@ -3,30 +3,131 @@
...
+@ExtendWith(MockitoExtension.class)
 public class AccountMapperTest {
 
-   // @Mock
+   @Mock
     private FetchToken testFetchToken = new FetchToken();
-   // @Mock
+   @Mock
     private AccountMapper testAccountMapper = new AccountMapper();
+    String battleTag = "Jokefish#2265";
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:27]: make all those private

File: src/main/java/com/diablo3CharViewer/token/FetchToken.java (hunk 2 above)
Comment (Arturnix) [reply src/main/java/com/diablo3CharViewer/token/FetchToken.java]: Kolizja nazw. Bohater jako gracz o przypisanym battle tagu. 
Zmieniłem na "Gracz".

File: src/test/java/AccountMapperTest.java (hunk 6 above)
Comment (Arturnix) [reply src/test/java/AccountMapperTest.java:27]: I made all those private

File: src/test/java/AccountMapperTest.java (hunk 5 above)
Comment (Arturnix) [reply src/test/java/AccountMapperTest.java:102]: If I am using @Mock instances here i get errors:
AssertionFailedError: Expected java.lang.NullPointerException to be thrown, but nothing was thrown.
So I use separately regular instances of those classes. I moved them at the begginig of test class where variables are declared.

File: src/test/java/AccountMapperTest.java (hunk 4 above)
Comment (Arturnix) [reply src/test/java/AccountMapperTest.java:120]: If I am using @Mock instances here i get errors:
AssertionFailedError: Expected java.lang.NullPointerException to be thrown, but nothing was thrown.
So I use separately regular instances of those classes. I moved them at the begginig of test class where variables are declared.

File: src/test/java/AccountMapperTest.java (hunk 3 above)
Comment (Arturnix) [reply src/test/java/AccountMapperTest.java:88]: Now it is assertTrue checking does returned waring contains desired message if you use battleTag with required format missmatch.

File: src/test/java/AccountMapperTest.java [hunk 7]
```diff
@@ -1,32 +1,121 @@
...
-    public void correctFetchedAccountToDataModelTest() {
+    public void correctAccountFetchedToDataModel() {
+
+       Mockito.when(testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock)).thenReturn(accountDataModel);
+       Assertions.assertEquals(expectedBattleTag, testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock).getBattleTag());
+    }
+
+    @Test
+    public void correctParagonLevelFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:53]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 8]
```diff
@@ -1,32 +1,121 @@
...
+    {{
+            put("elites",1974);
+    }};
+    private final AccountDataModel accountDataModel = new AccountDataModel("Jokefish#2265", 1111, "Phantas Magoria",
+            heroes, 70, mapKills);
 
    @Test
-    public void correctFetchedAccountToDataModelTest() {
+    public void correctAccountFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:46]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 9]
```diff
@@ -1,32 +1,121 @@
...
+    private final String battleTagAsWrongFormat = "abc123";
+    private final String battleTagAsNull = null;
+    private final String battleTagAsDosentExist = "dsf#123";
+    private final String wrongBattleTagFormatWarning = "Niepoprawny format battleTag! Spróbuj ponownie.";
+    private final ArrayList<HeroDataModel> heroes = new ArrayList<>(Arrays.asList(
+            new HeroDataModel(1, "A", "barbarian"),
+            new HeroDataModel(2, "B", "crusader")
+    ));
+    private final Map<String, Integer> mapKills = new HashMap<String, Integer>()
+    {{
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:38]: Nitpick:formatting

File: src/test/java/AccountMapperTest.java [hunk 10]
```diff
@@ -1,32 +1,121 @@
...
+    @Test
+    public void correctParagonLevelFetchedToDataModel() {
+
+        Mockito.when(testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock)).thenReturn(accountDataModel);
+        Assertions.assertEquals(expectedParagonLevel, testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock).getParagonLevel());
+    }
+
+    @Test
+    public void correctGuildNameFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:60]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 11]
```diff
@@ -1,32 +1,121 @@
...
+        Mockito.when(testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock)).thenReturn(accountDataModel);
+        Assertions.assertEquals(expectedGuildName, testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock).getGuildName());
+    }
+
+    @Test
+    public void correctHeroesListSizeFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:67]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 12]
```diff
@@ -1,32 +1,121 @@
...
+        lenient().when(testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock)).thenReturn(accountDataModel);
+        Assertions.assertEquals(2, heroes.size());
+    }
+
+    @Test
+    public void correctHighestHardcoreLevelFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:74]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 13]
```diff
@@ -1,32 +1,121 @@
...
+        Mockito.when(testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock)).thenReturn(accountDataModel);
+        Assertions.assertEquals(70, testAccountMapperMock.fetchAccountToDataModel(battleTag, testFetchTokenMock).getHighestHardcoreLevel());
+    }
+
+    @Test
+    public void correctKillsFetchedToDataModel() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:81]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 14]
```diff
@@ -1,32 +1,121 @@
...
+        Mockito.when(testAccountMapperMock.generateRequest(battleTagAsWrongFormat, testFetchTokenMock))
+                .thenReturn("Niepoprawny format battleTag! Spróbuj ponownie.");
+
+        Assertions.assertTrue(testAccountMapperMock.generateRequest(battleTagAsWrongFormat, testFetchTokenMock).contains(wrongBattleTagFormatWarning));
+    }
+
+    @Test
+    public void providedBattleTagDoesntExistThrowsException() {
+
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:98]: Nitpick: whitespace

File: src/test/java/AccountMapperTest.java [hunk 15]
```diff
@@ -1,32 +1,121 @@
...
+        String expectedMessage = "Gracz o podanym battleTagu nie istnieje w swiecie Sanktuarium!";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
 
-        String battleTag = "Jokefish#2265";
+    @Test
+    public void providedBattleTagIsNullThrowsException() {
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:111]: Nitpick: whitespace

**Pull request merged**